src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1211,6 +1237,23 @@ def forward(self, inputs, targets):
         else:
             raise UnsupportedOptimizerFormat()
 
+        if self.lr_mode is not None:
+            lr_scheduler_callback = create_lr_scheduler_callback(
+                lr_mode=self.lr_mode,
+                train_loader=self.train_loader,
+                net=self.net,
+                training_params=self.training_params,
+                update_param_groups=self.update_param_groups,
+                optimizer=self.optimizer,
+            )
+            self.phase_callbacks.append(lr_scheduler_callback)
+
+            # NEED ACCESS TO THE UNDERLYING TORCH SCHEDULER FOR LOADING/SAVING IT'S STATE_DICT
+            if isinstance(lr_scheduler_callback, LRSchedulerCallback):
+                self._torch_lr_scheduler = lr_scheduler_callback.scheduler
```

Comment:
What happes to the previous value of the `self._torch_lr_scheduler`?
What if it was None and become a non-None? What side effects could be happening here?

Reply:
It is only initiated with None.
Then, if we are using a torch scheduler it is set to it here.
So, the only time it would not be None, is if we call train() twice, and then we need to point to the right scheduler.